package/com.unity.formats.usd/Runtime/Scripts/IO/Geometry/InstanceImporter.cs

```diff
@@ -117,36 +123,45 @@ public static void BuildSceneInstances(PrimMap primMap, SceneImportOptions optio
                 GameObject goMaster;
                 if (!primMap.TryGetValue(new pxr.SdfPath(targetPath), out goMaster))
                 {
-                    Debug.LogWarning("Proto not found in PrimMap: " + targetPath);
                     continue;
                 }
 
-                if (i >= transforms.Length)
+                if (instanceCount >= transforms.Length)
                 {
-                    Debug.LogWarning("No transform for instance index [" + i + "] " +
+                    Debug.LogWarning("No transform for instance index [" + instanceCount + "] " +
                         "for instancer: " + pointInstancerPath);
                     break;
                 }
 
-                var xf = transforms[i];
+                var xf = transforms[instanceCount];
 
-                var instanceName = $"{goMaster.name}_{i}";
+                var instanceName = $"{goMaster.name}_{instanceCount}";
 
                 // If the old instance exists, we must destroy it to avoid a duplicate
-                // because the prototypes may have changed during re-import
+                // because the prototypes may have changed during re-import.
+                // The Transform hierarchy was created in the same order as it is now deleted,
+                // so this Find() operation is best-case performance.
                 var existingInstance = root.transform.Find(instanceName);
                 if (existingInstance != null)
                 {
                     GameObject.DestroyImmediate(existingInstance.gameObject);
                 }
 
-                var goInstance = GameObject.Instantiate(goMaster, root.transform);
+                instancesToCreate[instanceCount] = new Tuple<Matrix4x4, string, GameObject>(xf, instanceName, goMaster);
```

Comment:
can we have the tuple here named?
so that we dont have to use Item1, Item2, Item3 when referencing them?

Reply:
TIL about ValueTuples that can actually have named fields 😁 Done